### PR TITLE
Bare deployment for `stm32f103` board

### LINKED ISSUE #48: `deploy fishstick-stm32f401` run configuration has a broken pre-run task
It is configured to use `Build STM3F401` but that no longer exists, replaced with `build fishstick-stm32f401`

### PR DESCRIPTION
**Pull request by BryceBeagle**
Resolves #46
Resolves #48

### CODE REVIEW

**Review by rhedgeco — APPROVED** [review]
Looks good

File: .idea/runConfigurations/build_fishstick_stm32f103.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <option name="emulateTerminal" value="false" />
+    <option name="withSudo" value="false" />
+    <option name="buildTarget" value="REMOTE" />
+    <option name="backtrace" value="SHORT" />
+    <envs />
+    <option name="isRedirectInput" value="false" />
+    <option name="redirectInputPath" value="" />
+    <method v="2" />
+  </configuration>
+</component>
```
Comment (rhedgeco) [inline .idea/runConfigurations/build_fishstick_stm32f103.xml:17]: do these files not get newlines automatically?

File: .idea/runConfigurations/deploy_fishstick_stm32f103.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="deploy fishstick-stm32f103" type="com.jetbrains.cidr.embedded.openocd.conf.type" factoryName="com.jetbrains.cidr.embedded.openocd.conf.factory" REDIRECT_INPUT="false" ELEVATE="false" USE_EXTERNAL_CONSOLE="false" PASS_PARENT_ENVS_2="true" PROJECT_NAME="fishstick" TARGET_NAME="Do Nothing" CONFIG_NAME="Do Nothing" version="1" RUN_PATH="$PROJECT_DIR$/software/fishstick-stm32f103/target/thumbv7m-none-eabi/debug/fishstick-stm32f103">
+    <openocd version="1" gdb-port="3333" telnet-port="4444" board-config="$PROJECT_DIR$/software/fishstick-stm32f103/.openocd/stlink-stm32f1x.cfg" reset-type="RUN" download-type="ALWAYS">
+      <debugger kind="GDB" isBundled="true" />
+    </openocd>
+    <method v="2">
+      <option name="RunConfigurationTask" enabled="true" run_configuration_name="build fishstick-stm32f103" run_configuration_type="CargoCommandRunConfiguration" />
+    </method>
+  </configuration>
+</component>
```
Comment (rhedgeco) [inline .idea/runConfigurations/deploy_fishstick_stm32f103.xml:10]: newline here as well

File: .idea/runConfigurations/deploy_fishstick_stm32f401.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
       <debugger kind="GDB" isBundled="true" />
     </openocd>
     <method v="2">
-      <option name="RunConfigurationTask" enabled="true" run_configuration_name="Build STM3F401" run_configuration_type="CargoCommandRunConfiguration" />
+      <option name="RunConfigurationTask" enabled="true" run_configuration_name="build fishstick-stm32f401" run_configuration_type="CargoCommandRunConfiguration" />
     </method>
   </configuration>
 </component>
```
Comment (rhedgeco) [inline .idea/runConfigurations/deploy_fishstick_stm32f401.xml:10]: new... bum ba dum... line

**Review by rhedgeco — CHANGES_REQUESTED** [review]

File: .idea/runConfigurations/build_fishstick_stm32f103.xml (hunk 1 above)
Comment (BryceBeagle) [reply .idea/runConfigurations/build_fishstick_stm32f103.xml:17]: No, and it's pretty annoying. They're auto-generated without newlines and if you manually add them the IDE will remove them again the next time the configuration changes

Comment (rhedgeco) [reply .idea/runConfigurations/build_fishstick_stm32f103.xml:17]: even the newline on save setting in the IDE doesn't put it?

Comment (BryceBeagle) [reply .idea/runConfigurations/build_fishstick_stm32f103.xml:17]: They actually fight. If I manually make changes to the file (as text), it adds a new line. Then if I change the configuration through the UI, it removes the new line

Comment (rhedgeco) [reply .idea/runConfigurations/build_fishstick_stm32f103.xml:17]: okay well this is resolved then and we will forget about it lol

**Review by rhedgeco — APPROVED** [review]

**Pull request merged**